slash-commands/moderation/kick.js

```diff
@@ -0,0 +1,92 @@
+const Discord = require('discord.js');
+const {SlashCommandBuilder} = require('@discordjs/builders');
+const {promisePool} = require('../../config/db');
+const {sendToAuditLogsChannel} = require('../../helpers/sendToAuditLogs');
+const {
+  isServerStaff,
+  sendNoTargetStaffReply,
+} = require('../../helpers/validation');
+const {verifyReasonLength} = require('../../helpers/stringHelpers');
+
+const kickIntro = "You've been kicked for the following reason: ```";
+const kickOutro = ' ```';
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('kick')
+    .setDefaultPermission(false)
+    .setDescription('Kick a user')
+    .addUserOption((option) =>
+      option.setName('target').setDescription('The user').setRequired(true)
+    )
+    .addStringOption((option) =>
+      option.setName('reason').setDescription('The reason').setRequired(true)
+    ),
+  async execute(interaction) {
+    const targetUser = await interaction.guild.members.cache.get(
+      await interaction.options.getUser('target')?.id
+    );
+    if (!targetUser) {
+      return interaction.reply('The specified user could not be found.');
+    }
+
+    if (await isServerStaff(interaction, targetUser)) {
+      return await sendNoTargetStaffReply(interaction);
+    }
+
+    const reason = await interaction.options.getString('reason');
+    if (!verifyReasonLength(reason, interaction)) return;
+
+    const kickText = kickIntro + reason + kickOutro;
+
+    try {
+      await dmUser(interaction, targetUser, kickText);
+      await targetUser.kick(kickText);
+      await interaction.channel.send(`${targetUser} was kicked`);
+      await recordKickInDB(interaction, targetUser, reason);
+      await sendToAuditLogsChannel(interaction, {
+        color: '#0099ff',
+        titleMsg: `${targetUser.user.username}#${targetUser.user.discriminator} was kicked by ${interaction.user.tag}:`,
+        description: reason,
+        targetUser: targetUser.user,
+      });
+      await interaction.reply(`Kick command was successful`);
+    } catch (err) {
+      console.error(err);
+      return await interaction.reply(
+        `There was an error while executing the command`
+      );
+    }
+  },
+};
+
+async function dmUser(interaction, targetUser, reason) {
+  try {
+    await targetUser.send(reason);
+    await interaction.channel.send(
+      `Message regarding kick was sent to ${targetUser} successfully`
+    );
+  } catch (err) {
+    if (err.code === Discord.Constants.APIErrors.CANNOT_MESSAGE_USER) {
+      interaction.channel.send('The user cannot be messaged.');
+    } else {
+      throw new Error(err.message);
+    }
+  }
+}
+
+async function recordKickInDB(interaction, targetUser, reason) {
+  const kickSQL = `INSERT INTO infractions (timestamp, user, action, length_of_time, reason, valid, moderator) VALUES 
+  (now(), ?, 'cc!kick', NULL, ?, true, ?)`;
```

Comment:
Change `action` from `cc!kick` to `kicked [user ID]` to be consistent with other slash commands.

Reply:
Since other commands such as timeout just has the action (such as "timeout") in this column I modified the action to "kick". Both the user and moderator are covered in other columns of the table.